### DISCUSSION TITLE
Help resizing TextLayer font size

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#7838] [Q&A] [answered]
Hi, I'm working on a heatmap (link to [ObservableHQ notebook](https://observablehq.com/d/560fa52b8442ae63)) and I would like to have row label text that resizes based on zooming in or out. Currently I'm stuck getting the row label size to change at all when running the onViewStateChange function. 

![row_label_gif](https://user-images.githubusercontent.com/8352840/232273288-effc7d30-d9be-4ec3-9ae6-298a42d148d9.gif)

Here is a piece of code showing where I'm trying to set up a function `getSize` to re-run when a variable `zoom_trigger` is updated when the `on_view_state_change` function is run.  I can see the console log output from the "on_view_state" function running but I can't see the console log from the getSize function. 

```
{

  // this is being used for toggling the zoom state of the main matrix
  // var ini_zoom_mode = 'Y'
  // probably want to set up mutable variables to be initialize outside of this block of code

  let globalViewState = ini_global_view_state()

  // initialize zoom_data
  // zoom_data keeps track of the state of the zoom
  // I do not want to have to deal with a mutable zoom_data defined in a 
  // separate ObservableHQ cell so I'll just define it here
  var zoom_data = {
    pan_x: ini_x,
    pan_y: ini_y, 
    zoom_x: ini_zoom_x, 
    zoom_y: ini_zoom_y,
  }

  var ini_font_size = 10
  var zoom_trigger = 10;

  var on_view_state_change = ({viewState, viewId}) => {

    zoom_trigger++;  // Increment the zoom trigger every time the view state changes
    
    const {zoom, target, offset} = viewState;
    
    globalViewState = redefine_global_view_state(zoom_data, viewId, zoom, target)

    console.log(zoom_trigger)
    
    update_zoom_data(zoom_data, viewId, zoom, target)

    deckgl.setProps({viewState: {...globalViewState}});

  }  

  var row_label_layer  = new deck.TextLayer({
      id: 'row-label-layer',
      data: row_label_data, // This should be the same data source you used for your other layers
      getPosition: d => d.position,
      getText: d => d.name, // Replace 'label' with the property in your data that contains the text you want to display
      getSize: d => {
        console.log('get size', zoom_trigger)
        return zoom_trigger
      },        
      getColor: [0, 0, 0], // Text color as an RGBA array
      getAngle: 0, // Optional: Text angle in degrees
      getTextAnchor: 'end', // middle
      getAlignmentBaseline: 'center',
      fontFamily: 'Arial',
      sizeUnits: 'pixels',
      updateTriggers: {
        getSize: zoom_trigger
      }
    });
  
  const deckgl = new deck.DeckGL({
    container,  
    views,
    layerFilter, 
    layers: [mat_layer, row_cat_layer, col_cat_layer, row_label_layer],
    getTooltip: ({object}) => object && object.message,    
    initialViewState: globalViewState,
    onViewStateChange: on_view_state_change,
  })
  
}

```

I am working in the Orthographic projection and I would like to have fine grained control (specific control over row and column label font sizes), which is I was hesitant to take the "meters" unit approach to getting the font size to scale (my experiments with that seemed to work for reducing the initial font size but not increasing the font size when zooming).

**Comment by Pessimistress** [comment, marked as answer]
You need to create a new `TextLayer` instance and call `deckgl.setProps({layers})` when you want to update the size.

**Reply by cornhundred** [reply]
thanks, I was able to get it working on ObservableHQ in this [notebook](https://observablehq.com/d/1e3db0bae8d9afcf)
